src/lib/helpers.ts

```diff
@@ -315,3 +310,38 @@ export function getMaxBlockByChainId(chainId: number, targetBlock: number) {
   // In mainnet this is always the targetBlock, but in Goerli we add 24 blocks to the targetBlock.
   return targetBlock + env.chainIdBlockOffsets[chainId];
 }
+export class PrivateKeyManager {
+  private static instance: PrivateKeyManager;
+  private keys: Record<string, string> = {};
+
+  private constructor() { }
+
+  public static getInstance(): PrivateKeyManager {
+    if (!PrivateKeyManager.instance) {
+      PrivateKeyManager.instance = new PrivateKeyManager();
+    }
+    return PrivateKeyManager.instance;
+  }
+
+  public async initialize(ovalConfigs: OvalConfigs) {
```

Comment:
I've proposed to have an initialisation phase where all the gcmks/keys are stored in memory so it doesn't add any extra delays to the unlocks. What do you think?